astroquery/neos/core.py

```diff
@@ -0,0 +1,132 @@
+import re
+
+from bs4 import BeautifulSoup
+
+from astropy.time import Time
+from astropy.table import Table
+
+from . import conf
+from ..query import BaseQuery
+from ..utils import async_to_sync
+
+DEFAULT_API_KEY = "DEMO_KEY"
+
+
+@async_to_sync
+class NeowsClass(BaseQuery):
+    """
+    Class for querying NEOs orbit from NEOWS and JPL SBDB.
+    """
+
+    def __init__(self, api_key=None, spk_id=None, name=None):
+        """
+        Parameters:
+        ----------
+        api_key : str
+            NASA OPEN APIs key (default: `DEMO_KEY`)
+        """
+        super(NeowsClass, self).__init__()
+        self.spk_id = spk_id
+        self.name = name
+        self.TIMEOUT = conf.timeout
+        self.api_key = api_key
+
+    def _spk_id_from_name(self, name):
+        payload = {
+            "sstr": name,
+            "orb": "0",
+            "log": "0",
+            "old": "0",
+            "cov": "0",
+            "cad": "0",
+        }
+        SBDB_URL = conf.sbdb_server
+        response = self._request(
+            "GET", SBDB_URL, params=payload, timeout=self.TIMEOUT)
```

Comment:
Do you want to (optionally?) enable `cache`ing here?

Reply:
Yes, sure. I will do that!